sigmf/archive.py

```diff
@@ -21,80 +23,83 @@
 
 
 class SigMFArchive():
-    """Archive a SigMFFile.
+    """Archive one or more `SigMFFile`s.
 
     A `.sigmf` file must include both valid metadata and data.
     If `self.data_file` is not set or the requested output file
     is not writable, raise `SigMFFileError`.
 
     Parameters:
 
-      sigmffile -- A SigMFFile object with valid metadata and data_file
+      sigmffile -- An iterable of SigMFFile objects with valid metadata and data_files
 
       name      -- path to archive file to create. If file exists, overwrite.
-                   If `name` doesn't end in .sigmf, it will be appended.
-                   For example: if `name` == "/tmp/archive1", then the
-                   following archive will be created:
-                       /tmp/archive1.sigmf
-                       - archive1/
-                         - archive1.sigmf-meta
-                         - archive1.sigmf-data
+                    If `name` doesn't end in .sigmf, it will be appended. The
+                    `self.path` instance variable will be updated upon
+                    successful writing of the archive to point to the final
+                    archive path.
+
 
       fileobj   -- If `fileobj` is specified, it is used as an alternative to
-                   a file object opened in binary mode for `name`. It is
-                   supposed to be at position 0. `name` is not required, but
-                   if specified will be used to determine the directory and
-                   file names within the archive. `fileobj` won't be closed.
-                   For example: if `name` == "archive1" and fileobj is given,
-                   a tar archive will be written to fileobj with the
-                   following structure:
-                       - archive1/
-                         - archive1.sigmf-meta
-                         - archive1.sigmf-data
+                    a file object opened in binary mode for `name`. If
+                    `fileobj` is an open tarfile, it will be appended to. It is
+                    supposed to be at position 0. `fileobj` won't be closed. If
+                    `fileobj` is given, `name` has no effect.
     """
-    def __init__(self, sigmffile, name=None, fileobj=None):
-        self.sigmffile = sigmffile
+    def __init__(self, sigmffiles : Iterable["SigMFFile"], name : str = None, fileobj : BinaryIO =None):
```

Comment:
The inclusion of the `archive` function in `SigMFFile` helps alleviate this a lot, but I think this would still be an easy change to make the syntax more intuitive for the single-recording archive case.

Reply:
Actually, I thought of something to try

Reply:
fixed in 832b731

Reply:
I tried this approach, which seems to work. It doesn't follow the same style for local imports within the various modules of the package, but this approach seems to avoid the circular import problem:

`archive.py`:
```python
from typing import Iterable
import sigmf.sigmffile as sigmffile

class SigMFArchive:
    def __init__(self, file_or_files, ...):
        if isinstance(file_or_files, sigmffile.SigMFFile):
            print("This is a single, iterable SigMF file")
        elif isinstance(file_or_files, Iterable):
            print("This is another iterable")
        ...
```

`sigmffile.py`:
```python
import sigmf.archive as archive
class SigMFFile:
    ...
    def archive(self, ...):
        # this variable cannot be named "archive" due to the import
        archived = archive.SigMFArchive(self, ...)
        return archived
    ...
```

Reply:
There was bug with the implementation I had, so I changed it to be more similar to your and Doug Anderson's versions in 4f58453